sources/CGFloat+HUBMath.h

```diff
@@ -0,0 +1,139 @@
+/*
+ *  Copyright (c) 2016 Spotify AB.
+ *
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+#import <CoreGraphics/CoreGraphics.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+#pragma mark Computing Values from CGFloats
+
+/**
+ Round to largest integral value not greater than `value`.
+ 
+ @note Consider using this function instead of the standard `floor` and `floorf` functions for `CGFloat` values. This
+       function helps avoid implicit double promotion and loss of precision. Keeping the types correct regardless of
+       if the backing type is `double` or `float`.
+
+ @param value The value that should be floored.
+ @return The largest integral value not greater than `value`.
+ */
+static inline CGFloat HUBCGFloatFloor(CGFloat value)
+{
+#if CGFLOAT_IS_DOUBLE
+    return floor(value);
+#else
+    return floorf(value);
+#endif // CGFLOAT_IS_DOUBLE
```

Comment:
Can we use `tgmath.h` or C11 generic macros instead of our own helper functions?